vm/src/pyobject.rs

```diff
@@ -941,51 +913,40 @@ impl<T> TypeProtocol for PyRef<T> {
 }
 
 pub trait DictProtocol {
```

Comment:
I'm in favor of removing this trait instead of the item trait. In CPython its called `Item protocol`.

Reply:
That's roughly the direction I'm going in. There's a couple of tricky points:

Do we implement ItemProtocol on PyDictRef and if so should it be the full magic method powered version or the direct implementation.

If the full version, how do we name the underlying implementation.

Reply:
If you have `PyDictRef` there is not need to go via magic methods (performance win!). If you have `PyObjectRef` you go the magic method path. We can implement `ItemProtocol` twice. Once on `PyObjectRef` and once on `PtDictRef`.

Reply:
The problem is that for subclasses of dict the behaviour is different. It's liable to lead to very subtle bugs in ```dict.into_object().get_item``` and ```dict.get_item``` do different things.